### PR TITLE
create #938_Range_Sum_of_BST.py

### LINKED ISSUE #76: #938_Range_Sum_of_BST
# Summary
add 938. Range Sum of BST

# Goal
add 938. Range Sum of BST

# Todo
- [ ] add 938. Range Sum of BST
- [ ] time complexity: O(logn)
- [ ] space complexity: O(n)

# Deadline
/

# Parent issue
None

# References
None

# Notes
None

### PR DESCRIPTION
**Pull request by ashizawaN**
Closes #76 

I thought the only way to solve this problem was to use a binary search tree, so I used only one solution method.
Please check my code.

### CODE REVIEW

File: #938_Range_Sum_of_BST.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+class Solution:
+    def rangeSumBST(self, root: TreeNode, low: int, high: int) -> int:
+        if not root:
+            return 0
+        else:
```
Comment (nayuta-ai) [inline #938_Range_Sum_of_BST.py:12]: You may want to consider using "early return".

https://zenn.dev/media_engine/articles/early_return

**Review by nayuta-ai — CHANGES_REQUESTED** [review]

File: #938_Range_Sum_of_BST.py (hunk 1 above)
Comment (ashizawaN) [reply #938_Range_Sum_of_BST.py]: I rewrote the code using it. Is this writing correct?

File: #938_Range_Sum_of_BST.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class TreeNode:
+    def __init__(self, val=0, left=None, right=None):
```
Comment (nayuta-ai) [inline #938_Range_Sum_of_BST.py:2]: Add the type annotation.

File: #938_Range_Sum_of_BST.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+class Solution:
+    def rangeSumBST(self, root: TreeNode, low: int, high: int) -> int:
```
Comment (nayuta-ai) [inline #938_Range_Sum_of_BST.py:9]: def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:

**Review by nayuta-ai — CHANGES_REQUESTED** [review]

**Review by nayuta-ai — APPROVED** [review]

**Pull request merged**